From ce5c83496ff344e61d9c9e875b6ca4f480612c1e Mon Sep 17 00:00:00 2001
From: Cazen Lee <Cazen.lee@samsung.com>
Date: Wed, 16 Sep 2015 01:04:00 -0800
Subject: [PATCH 0398/1164] CDH-34523: HIVE-11825 : 
 get_json_object(col,'$.a') is null in where
 clause didnt work (Cazen Lee via Ashutosh
 Chauhan)

Change-Id: I5b0ae38ec9cb05e4a5df5d70ebf354be9f66e606
Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
---
 .../org/apache/hadoop/hive/ql/udf/UDFJson.java     |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFJson.java b/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFJson.java
index 66778e0..907e807 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFJson.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFJson.java
@@ -66,6 +66,8 @@
   static {
     // Allows for unescaped ASCII control characters in JSON values
     JSON_FACTORY.enable(Feature.ALLOW_UNQUOTED_CONTROL_CHARS);
+    // Enabled to accept quoting of all character backslash qooting mechanism
+    JSON_FACTORY.enable(Feature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER);
   }
   private static final ObjectMapper MAPPER = new ObjectMapper(JSON_FACTORY);
   private static final JavaType MAP_TYPE = TypeFactory.fromClass(Map.class);
-- 
1.7.9.5

